core/trino-main/src/main/java/io/trino/connector/CatalogStoreConfig.java

```diff
@@ -18,21 +18,16 @@
 
 public class CatalogStoreConfig
 {
-    public enum CatalogStoreKind
-    {
-        MEMORY, FILE
-    }
-
-    private CatalogStoreKind catalogStoreKind = CatalogStoreKind.FILE;
+    private String catalogStoreKind = "file";
 
     @NotNull
-    public CatalogStoreKind getCatalogStoreKind()
+    public String getCatalogStoreKind()
```

Comment:
the actual value doesn't seem to matter. If the value is anything OTHER than `file` or `memory` the `CatalogStoreManager` gets bound which looks at `etc/catalog-store.properties` to figure out what actual `CatalogStore` to use. See https://github.com/trinodb/trino/blob/76ee82022390b9ac4b854841b78eb52e0cbbaa04/core/trino-main/src/main/java/io/trino/connector/DynamicCatalogManagerModule.java#L39

I propose to reintroduce `CatalogStoreKind` with `CUSTOM` as a value for the custom catalogstore impl.